src/util/crypto.ts

```diff
@@ -22,6 +24,33 @@ export class CryptoUtil {
     return ((await db.collection(Collections.config).get()).docs[0].data() as Config).jwt.secret;
   }
 
+  // Everything together: get token, verify token and auth token.
+  // Auth is in most cases seperate but for sake of simplicity I don't care.
+  static async verifyToken(req: Request, res: Response, next: NextFunction): Promise<void> {
+    const header = req.headers.authorization;
+    if (header != null) {
+      const token = header.split(' ')[1];
+
+      jwt.verify(token, await CryptoUtil.getSecret(), (err, authData) => {
```

Comment:
maybe use returns instead of if else chains

Reply:
Wdym returns? There's only send and next(). Is fine for now I think